File: 228. Summary Ranges/228.go
package leetcode
import(
	"strconv"
)

func summaryRanges(nums []int) []string {
    if len(nums) == 0{
		return []string{}
	}
	
	start := nums[0]
	last := nums[0]
	res := make([]string, 0, 1)

	for i:=1; i<len(nums); i++{
		if nums[i] == last+1{
			last = nums[i]
		}else{
			if start == last{
				res = append(res, strconv.Itoa(start))
			}else{
				res = append(res, strconv.Itoa(start)+"->"+strconv.Itoa(last))
			}
			start = nums[i]
			last = nums[i]
		}
	}
	if start == last{
		res = append(res, strconv.Itoa(start))
	}else{
		res = append(res, strconv.Itoa(start)+"->"+strconv.Itoa(last))
	}
	return res
}

File: 228. Summary Ranges/_228.py
class Solution:
    def summaryRanges(self, nums):
        if not nums:
            return []
        start = nums[0]
        last = nums[0]
        res = list()
        for i in nums[1:]:
            if i == last+1:
                last = i
            else:
                if last == start:
                    res.append(str(start))
                else:
                    res.append(str(start)+'->'+str(last))
                start = i
                last = i

        if last == start:
            res.append(str(start))
        else:
            res.append(str(start)+'->'+str(last))

        return res

    def summaryRanges1(self, nums):
        ranges = []
        for n in nums:
            if not ranges or n > ranges[-1][-1] + 1:
                ranges += [],
            ranges[-1][1:] = n,
        return ['->'.join(map(str, r)) for r in ranges]